Expensify/App
# Edit Comments

Pullerbearing (@MariaHCD)
cc @roryabraham 

### Details
Give users the ability to edit their comments!

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/147482
Fixes https://github.com/Expensify/Expensify/issues/162717

### Tests/QA
1. Open up a chat
1. On Web/Desktop, click the pencil button that shows when you hover over a chat message of your own. On Mobile, long tap a message of your own
1. Make sure that the comment shows up inside of the text input with the "Cancel" and "Save" buttons underneath
1. Edit the comment, click "Cancel" make sure that the original comment still shows
1. Edit the comment, click "Save" make sure that the comment shows as what you've edited it to become, with `(edited)` showing up next to the comment now

---

1.  Hover over a comment that is not your own, make sure that no "pencil" edit icon shows up and that you cannot edit that comment

---

### Tested On

- [ x ] Web
- [ x ] Mobile Web
- [ x ] Desktop
- [ x ] iOS

### Screenshots

#### Web
![Kapture 2021-04-30 at 15 53 18](https://user-images.githubusercontent.com/4741899/116761436-808bcc00-a9cc-11eb-94d6-7a1f51d6b6c4.gif)

#### Mobile Web
![Kapture 2021-04-30 at 16 01 52](https://user-images.githubusercontent.com/4741899/116761786-67cfe600-a9cd-11eb-8029-d7cbcd65279f.gif)

#### Desktop
![Kapture 2021-04-30 at 16 31 52](https://user-images.githubusercontent.com/4741899/116763159-b8e1d900-a9d1-11eb-9ab8-2d40f41e5070.gif)

#### iOS
![Kapture 2021-04-30 at 16 35 25](https://user-images.githubusercontent.com/4741899/116763286-1ece6080-a9d2-11eb-9a40-c7e42f19df4b.gif)

#### Android
I can't get the Android emulator to run without slowing my machine to a crawl so animations, so here are some stills:
![Screen Shot 2021-04-30 at 4 48 57 PM](https://user-images.githubusercontent.com/4741899/116763932-2131ba00-a9d4-11eb-8820-f82ac8b5b26c.png)
![Screen Shot 2021-04-30 at 4 49 23 PM](https://user-images.githubusercontent.com/4741899/116763944-2ee73f80-a9d4-11eb-8873-54dc014c8c04.png)
![Screen Shot 2021-04-30 at 4 49 52 PM](https://user-images.githubusercontent.com/4741899/116763934-2262e700-a9d4-11eb-822c-343bd7744deb.png)
